pkg/webhook/pod/mutating/cluster_colocation_profile.go

```diff
@@ -0,0 +1,209 @@
+	if len(profile.Spec.Labels) > 0 {
+		if len(pod.Labels) == 0 {
+			pod.Labels = make(map[string]string)
+		}
+		for k, v := range profile.Spec.Labels {
+			pod.Labels[k] = v
+		}
+	}
+
+	if len(profile.Spec.Annotations) > 0 {
+		if len(pod.Annotations) == 0 {
+			pod.Annotations = make(map[string]string)
+		}
+		for k, v := range profile.Spec.Annotations {
+			pod.Annotations[k] = v
+		}
+	}
+
+	if profile.Spec.SchedulerName != "" {
+		pod.Spec.SchedulerName = profile.Spec.SchedulerName
+	}
+
+	if profile.Spec.Patch.Raw != nil {
+		cloneBytes, _ := json.Marshal(pod)
+		modified, err := strategicpatch.StrategicMergePatch(cloneBytes, profile.Spec.Patch.Raw, &corev1.Pod{})
+		if err != nil {
+			return err
+		}
+		newPod := &corev1.Pod{}
+		if err = json.Unmarshal(modified, newPod); err != nil {
+			return err
+		}
+		*pod = *newPod
+	}
+
+	return nil
+}
+
+func (h *PodMutatingHandler) mutatePodResourceSpec(pod *corev1.Pod) error {
+	priorityClass := extension.GetPriorityClass(pod)
+	if priorityClass == extension.PriorityNone || priorityClass == extension.PriorityProd {
+		return nil
+	}
+
+	for _, containers := range [][]corev1.Container{pod.Spec.InitContainers, pod.Spec.Containers} {
+		for i := range containers {
+			container := &containers[i]
+			replaceAndEraseResource(priorityClass, container.Resources.Requests, corev1.ResourceCPU)
+			replaceAndEraseResource(priorityClass, container.Resources.Requests, corev1.ResourceMemory)
+
+			replaceAndEraseResource(priorityClass, container.Resources.Limits, corev1.ResourceCPU)
+			replaceAndEraseResource(priorityClass, container.Resources.Limits, corev1.ResourceMemory)
+
+			restrictResourceRequestAndLimit(priorityClass, container.Resources.Requests, container.Resources.Limits, corev1.ResourceCPU)
+			restrictResourceRequestAndLimit(priorityClass, container.Resources.Requests, container.Resources.Limits, corev1.ResourceMemory)
+		}
+	}
+
+	if pod.Spec.Overhead != nil {
+		replaceAndEraseResource(priorityClass, pod.Spec.Overhead, corev1.ResourceCPU)
+		replaceAndEraseResource(priorityClass, pod.Spec.Overhead, corev1.ResourceMemory)
+	}
+	return nil
+}
+
+func replaceAndEraseResource(priorityClass extension.PriorityClass, resourceList corev1.ResourceList, resourceName corev1.ResourceName) {
+	extendResourceName := extension.ResourceNameMap[priorityClass][resourceName]
+	if extendResourceName == "" {
+		return
+	}
+	quantity, ok := resourceList[resourceName]
+	if ok {
+		resourceList[extendResourceName] = quantity
+		delete(resourceList, resourceName)
+	}
+}
+
+func restrictResourceRequestAndLimit(priorityClass extension.PriorityClass, requests, limits corev1.ResourceList, resourceName corev1.ResourceName) {
+	extendResourceName := extension.ResourceNameMap[priorityClass][resourceName]
+	if extendResourceName == "" {
+		return
+	}
+	requestQuantity, requestOK := requests[extendResourceName]
+	limitQuantity, limitOK := limits[extendResourceName]
+	if requestOK && !limitOK {
```

Comment:
According to the habit of kubernetes, remains unchanged when the limit is not set?